_includes/pbc/all/install-features/202304.0/install-push-notification-feature.md

```diff
@@ -349,19 +349,284 @@ class GlueBackendApiApplicationDependencyProvider extends SprykerGlueBackendApiA
+          changePushButtonState('disabled');
+        })
+        .then(subscription => subscription.unsubscribe())
+        .then(() => changePushButtonState('disabled'))
+        .catch(e => {
+          // We failed to unsubscribe, this can lead to
+          // an unusual state, so  it may be best to remove
+          // the users data from your data store and
+          // inform the user that you have done so
+          console.error('Error when unsubscribing the user', e);
+          changePushButtonState('disabled');
+        });
+  }
+
+  async function getToken() {
+    const response = await fetch('http://glue-backend.de.spryker.local/token', {
+      method: "POST",
+      body: JSON.stringify({
+        grantType: "password",
+        username: "",
+        password: ""
+      }),
+    });
+
+    return response.json()
+  }
+
+  async function push_sendSubscriptionToServer(subscription) {
+    const tokenScos = await getToken();
+    const accessToken = tokenScos[0].access_token;
+    const key = subscription.getKey('p256dh');
+    const token = subscription.getKey('auth');
+
+    return fetch('http://glue-backend.de.spryker.local/push-notification-subscriptions', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Authorization': `Bearer ${accessToken}`,
+        'Content-Type': 'application/vnd.api+json'
+      },
+      body: JSON.stringify({
+        data: {
+          type: "push-notification-subscriptions",
+          attributes: {
+            providerName: "web-push-php",
+            group: {
+              name: "warehouse",
+              identifier: 1234
+            },
+            'payload': {
+              endpoint: subscription.endpoint,
+              publicKey: key ? btoa(String.fromCharCode.apply(null, new Uint8Array(key))) : null,
+              authToken: token ? btoa(String.fromCharCode.apply(null, new Uint8Array(token))) : null,
+            }
+          }
+        }
+      }),
+    }).then(() => subscription);
+  }
+});
+```
+
+** .../push_notification_spa/serviceWorker.js **
+```js
+self.addEventListener('push', function (event) {
+    if (!(self.Notification && self.Notification.permission === 'granted')) {
+        return;
+    }
+
+    const sendNotification = body => {
+        const title = "Web Push example";
+
+        return self.registration.showNotification(title, {
+            body,
+        });
+    };
+
+    if (event.data) {
+        const message = event.data.text();
+        event.waitUntil(sendNotification(message));
+    }
+});
+```
+
+5. Setup credentials:
+   - Open `.../push_notification_spa/app.js` and replace the `applicationServerKey` variable value with your VAPID public key.
+   - Find the `getToken()` method at the `.../push_notification_spa/app.js` and replace the credentials with the user that works in your system. 
+6. Run the local http server with the demo app by executing following console command `php -S localhost:8000`.
```

Comment:
* Do we still need to run the local server or open the page in a browser would be enough?

Reply:
serverWorker registration is failing if page just opened in browser.
```
navigator.serviceWorker.register('serviceWorker.js').then(
      () => {
        console.log('[SW] Service worker has been registered');
      },
```

[SW] Service worker registration failed TypeError: Failed to register a ServiceWorker: The URL protocol of the current origin ('file://') is not supported.